src/nunit.analyzers/SameAsIncompatibleTypes/SameAsIncompatibleTypesAnalyzer.cs

```diff
@@ -60,7 +60,7 @@ public class SameAsIncompatibleTypesAnalyzer : BaseAssertionAnalyzer
                 {
                     context.ReportDiagnostic(Diagnostic.Create(
                         descriptor,
-                        expectedArgumentExpression.GetLocation()));
+                        expectedArgumentExpression!.GetLocation()));
```

Comment:
Line 42: 
`.Where(ex => ex != null)`
We know for sure that `expectedArgumentExpression` is not null, it's just compiler is not smart enough to know that as well.

Reply:
My fault, I confused the C# type system with other type systems - where it would propagate the non-null property. Of course the typing of `Where` does not change the type of the elements, as it is just a filtering.